app/src/main/java/org/p2p/wallet/infrastructure/network/data/transactionerrors/RpcTransactionErrorTypeAdapter.kt

```diff
@@ -0,0 +1,88 @@
+package org.p2p.wallet.infrastructure.network.data.transactionerrors
+
+import com.google.gson.TypeAdapter
+import com.google.gson.stream.JsonReader
+import com.google.gson.stream.JsonToken
+import com.google.gson.stream.JsonWriter
+import org.p2p.core.utils.nextArray
+import org.p2p.core.utils.nextObject
+
+/**
+ * For parsing structs like:
+ * ```json
+ * "err": {
+ *    "InstructionError": [
+ *        3,
+ *        { "Custom": 6022 }
+ *    ]
+ * }
+ *
+ *    or
+ * "err": {
+ *    "InstructionError": [
+ *        3,
+ *        "Other"
+ *    ]
+ * }
+ *    or
+ * "err": "AccountInUse"
+ * ```
+ * @see [RpcTransactionErrorParserTest]
+ * @return valid typed RpcTransactionError or null if parsing is failed
+ */
+class RpcTransactionErrorTypeAdapter(
+    private val instructionErrorParser: RpcTransactionInstructionErrorParser
+) : TypeAdapter<RpcTransactionError>() {
+
+    override fun read(reader: JsonReader): RpcTransactionError {
+        return try {
+            when (val errKeyType = reader.peek()) {
+                // : "AccountInUse"
+                JsonToken.STRING -> createPrimitiveError(reader.nextString())
+                // : { "DuplicateInstruction": [ 3 ] }
+                JsonToken.BEGIN_OBJECT -> reader.nextObject(::parseObjectError)
+                else -> RpcTransactionError.Unknown("Failed to parse error for $errKeyType in ${reader.path}")
+            }
+        } catch (parsingFatalError: Throwable) {
+            throw TransactionErrorParseFailed(
```

Comment:
Remove throw and replace it with Unknown. Better to log timber.e then crash the whole app